src/compton/test/t2Compton.cc

```diff
@@ -0,0 +1,322 @@
+            const size_t i = i3 + sz3 * (i2 + sz2 * (i1 + sz1 * (i0)));
+            flat_interp_data[i] = interp_data[i0][i1][i2][i3];
+          }
+        }
+      }
+    }
+
+    // Test sizes
+    ut.check(sz0 == num_evals_gold, "tested evals size");
+    ut.check((sz1 == num_groups_gold) && (sz2 == num_groups_gold), "tested groups size");
+    ut.check(sz3 == num_leg_moments_gold, "tested Legendre moments size");
+
+    // Print result (useful if golds need updating)
+    if (do_print) {
+      [](const std::vector<double> &v) {
+        auto print = [](double a) { std::cout << a << ", "; };
+        std::cout << std::setprecision(14);
+        std::cout << '\n';
+        std::for_each(v.begin(), v.end(), print);
+        std::cout << std::endl;
+      }(flat_interp_data);
+    }
+
+    // NB: These values come directly from the ASCII data files
+    std::vector<double> flat_interp_gold = {
+        // out_lin
+        1.023686968316, 0.003573655955675, 0.1408569058113, -0.003385368827333, 0, 0, 0, 0,
+        0.01368733137026, -0.000337289996441, 0.8580132416506, 0.00346237220459, 0.1150544176069,
+        -0.002695776219916, 0, 0, 0, 0, 0.03016927525143, -0.000745487878607, 0.7395544180756,
+        0.003537535339196, 0.08790804121747, -0.001974567477495, 0, 0, 0, 0, 0.02760683687763,
+        -0.0006747774819197, 0.7154380590368, 0.002500085526454,
+        // in_lin
+        1.029549163582, 0.003360819879047, 0.1682617986774, -0.004498320348364, 0, 0, 0, 0,
+        0.01155720790042, -0.000254680100335, 0.8615393494377, 0.003337305544177, 0.1370933267956,
+        -0.003577889661548, 0, 0, 0, 0, 0.02546236651279, -0.0005628307049739, 0.74122185377,
+        0.003480918190072, 0.104392775669, -0.002618892286625, 0, 0, 0, 0, 0.02330836203736,
+        -0.0005092109709329, 0.7158405069922, 0.002485525111488,
+        // out_nonlin
+        2.417350052736e-20, 5.333857003773e-23, 1.569215319107e-22, -3.474039416897e-24, 0, 0, 0, 0,
+        2.025140617993e-22, -5.27716088378e-24, 4.991731616824e-22, 1.576887628939e-24,
+        7.937214293507e-24, -1.661082680445e-25, 0, 0, 0, 0, 1.15611911544e-23, -3.08379217909e-25,
+        2.906510852958e-23, 1.161410570085e-25, 3.652307822369e-25, -6.9213492314e-27, 0, 0, 0, 0,
+        6.471151078044e-25, -1.788886184269e-26, 9.067645193159e-25, 3.288019868804e-27,
+        // in_nonlin
+        2.412582001649e-20, 5.506360234252e-23, 1.858984933376e-22, -4.598103855448e-24, 0, 0, 0, 0,
+        1.70028840385e-22, -3.978450052949e-24, 4.964897275093e-22, 1.671777660171e-24,
+        9.356135881693e-24, -2.194753402066e-25, 0, 0, 0, 0, 9.680369923644e-24, -2.32267253788e-25,
+        2.875946793361e-23, 1.265594484598e-25, 4.274264612141e-25, -9.142968009692e-27, 0, 0, 0, 0,
+        5.392751048597e-25, -1.344066881751e-26, 8.87128629256e-25, 3.986007245261e-27};
+
+    ut.check(std::equal(flat_interp_data.begin(), flat_interp_data.end(), flat_interp_gold.begin(),
```

Comment:
simplify with `soft_equiv(a.begin(), a.end(), b.begin(), ...)`

Reply:
what's the last argument? the tolerance?

Reply:
> what's the last argument? the tolerance?

Yes. 

```c++
template <typename Value_Iterator, typename Ref_Iterator>
inline bool
soft_equiv(Value_Iterator value, Value_Iterator value_end, Ref_Iterator ref, Ref_Iterator ref_end,
           typename std::iterator_traits<Value_Iterator>::value_type const precision = 1.0e-12);
```

or 

```c++
inline bool soft_equiv(const std::vector<double> &value, const std::vector<double> &ref,
                       const double precision = 1.0e-12
```